File: 09-01-shared-element/app/src/main/java/com/example/sharedElement/MainActivity.java
package com.example.sharedElement;

import android.app.ActivityOptions;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.ImageView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        final ImageView imageview = (ImageView) findViewById(R.id.imageView);
        imageview.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, DetailActivity.class);
                if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                    ActivityOptions activityOptions = ActivityOptions.makeSceneTransitionAnimation(MainActivity.this, imageview, "transitionImage");
                    startActivity(intent, activityOptions.toBundle());
                } else {
                    startActivity(intent);
                }
            }
        });
    }
}

File: 12-01-shared-element/kotlin-12-01-shared-element/app/src/main/java/com/example/sharedElement/MainActivity.kt
package com.example.sharedElement

import android.app.ActivityOptions
import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.ImageView

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val imageview = findViewById<ImageView>(R.id.imageView)
        imageview.setOnClickListener { v ->
            val intent = Intent(this@MainActivity, DetailActivity::class.java)
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                val activityOptions = ActivityOptions.makeSceneTransitionAnimation(this@MainActivity, imageview, "transitionImage")
                startActivity(intent, activityOptions.toBundle())
            } else {
                startActivity(intent)
            }
        }
    }
}
